cell_classification/segmentation_data_prep.py

```diff
@@ -102,47 +103,43 @@ def get_inst_binary_masks(self, data_folder):
         interior = np.logical_and(edge == 0, instance_mask > 0).astype(np.uint8)
         return interior, instance_mask
 
-    def get_cell_types(self, sample_name):
-        """Gets the cell types from the cell table for the given labels
-        Args:
-            sample_name (str):
-                The name of the sample we use for look up in column sample_key
-                in cell_type.csv
-        Returns:
-            df:
-                The labels and corresponding cell types for the given sample
-        """
-        return None
-
-    def get_marker_activity(self, cell_types, marker):
+    def get_marker_activity(self, sample_name, conversion_matrix, marker):
         """Gets the marker activity for the given labels
         Args:
-            cell_types list:
-                The cell types to get the marker activity for
+            sample_name (str):
+                The name of the sample
+            conversion_matrix (pd.DataFrame):
+                The conversion matrix to use for the lookup
             marker (str, list):
                 The markers to get the activity for
         Returns:
-            list:
+            np.array:
                 The marker activity for the given labels, 1 if the marker is active, 0
                 otherwise and -1 if the marker is not specific enough to be considered active
         """
-        return None
+        sample_subset = self.cell_type_table[self.cell_type_table.SampleID == sample_name]
+        cell_types = sample_subset[self.cell_type_key].values
+        return conversion_matrix.loc[cell_types, marker].values
 
-    def get_marker_activity_mask(self, instance_mask, cell_types, marker_activity):
+    def get_marker_activity_mask(self, instance_mask, binary_mask, marker_activity):
         """Makes a mask from the marker activity
 
         Args:
             instance_mask (np.array):
                 The instance mask to make the marker activity mask for
-            cell_types list:
-                The cell types of the cells in the instance_mask
+            binary_mask (np.array):
+                The binary mask to make the marker activity mask for
             marker_activity list:
                 The marker activity that maps to the cell types
         Returns:
             np.array:
                 The marker activity mask
         """
-        return np.zeros([500, 500])
+        out_mask = np.zeros_like(instance_mask)
+        for label, activity in enumerate(marker_activity, 1):
```

Comment:
We don't know that the segmentation masks are going to be sequentially labeled without missing values. 
The post-processing we use for Mesmer, for example, will remove cells below a certain threshold, but it won't relabel the mask. This means it could go [1, 2, 4, 5, 6, 8, etc..]. We should probably directly iterate over the labels, rather than assuming that the index is equivalent to the labels.